File: scratch/find_duplicates.js
const fs = require('fs');
const path = require('path');

const filePath = path.join(__dirname, '../admin.js');
const content = fs.readFileSync(filePath, 'utf-8');

// Regex to find functions
const funcRegex = /(?:async\s+)?function\s+(\w+)\s*\(/g;
let match;
const counts = {};

while ((match = funcRegex.exec(content)) !== null) {
    const name = match[1];
    counts[name] = (counts[name] || 0) + 1;
}

const duplicates = {};
for (const [key, value] of Object.entries(counts)) {
    if (value > 1) {
        duplicates[key] = value;
    }
}

console.log("Duplicate functions:", duplicates);

File: scratch/find_duplicates.py
import re

with open('../admin.js', 'r', encoding='utf-8') as f:
    content = f.read()

# Find all function declarations: function name(...) or async function name(...)
funcs = re.findall(r'(?:async\s+)?function\s+(\w+)\s*\(', content)

# Count occurrences
counts = {}
for func in funcs:
    counts[func] = counts.get(func, 0) + 1

duplicates = {k: v for k, v in counts.items() if v > 1}
print("Duplicate functions:", duplicates)
